### PR TITLE
gh-99741: Implement Multi-Phase Init for the _xxsubinterpreters Module

### PR DESCRIPTION
**Pull request by ericsnowcurrently**
`_xxsubinterpreters` is an internal module used for testing.

* Issue: gh-99741

### CODE REVIEW

File: Modules/_xxsubinterpretersmodule.c [hunk 1]
```diff
@@ -28,13 +31,147 @@ _copy_raw_string(PyObject *strobj)
...
+    assert(mod != NULL);
+    module_state *state = PyModule_GetState(mod);
+    assert(state != NULL);
+    return state;
+}
+
+static int
+traverse_module_state(module_state *state, visitproc visit, void *arg)
+{
+    /* interpreter exceptions */
```
Comment (corona10) [inline Modules/_xxsubinterpretersmodule.c:184]:
```suggestion
    /* heap type */
    Py_VISIT(state->ChannelIDType);
    
    /* interpreter exceptions */
```

File: Modules/_xxsubinterpretersmodule.c [hunk 2]
```diff
@@ -28,13 +31,147 @@ _copy_raw_string(PyObject *strobj)
...
+    Py_VISIT(state->ChannelClosedError);
+    Py_VISIT(state->ChannelEmptyError);
+    Py_VISIT(state->ChannelNotEmptyError);
+    return 0;
+}
+
+static int
+clear_module_state(module_state *state)
+{
+    /* interpreter exceptions */
```
Comment (corona10) [inline Modules/_xxsubinterpretersmodule.c:203]:
```suggestion
    /* heap type */
    Py_CLEAR(state->ChannelIDType);
    
    /* interpreter exceptions */
```

File: Modules/_xxsubinterpretersmodule.c (hunk 1 above)
Comment (ericsnowcurrently) [reply Modules/_xxsubinterpretersmodule.c:184]: fixed

File: Modules/_xxsubinterpretersmodule.c (hunk 2 above)
Comment (ericsnowcurrently) [reply Modules/_xxsubinterpretersmodule.c:203]: fixed

File: Modules/_xxsubinterpretersmodule.c [hunk 3]
```diff
@@ -1972,18 +2204,41 @@ _run_script_in_interpreter(PyInterpreterState *interp, const char *codestr,
...
 }
 
+static void
+_globals_fini(void)
+{
+    // XXX This isn't thread-safe.
+    if (_globals.module_count > 1) {
+        return;
+    }
+    _globals.module_count--;
```
Comment (erlend-aasland) [inline Modules/_xxsubinterpretersmodule.c]: `module_count` will never reach 0. You'll need to do something like this:

```suggestion
    if (_globals.module_count-- > 1) {
        return;
    }
```

Comment (ericsnowcurrently) [reply Modules/_xxsubinterpretersmodule.c]: good catch!

Comment (ericsnowcurrently) [reply Modules/_xxsubinterpretersmodule.c]: fixed

**Pull request merged**